dual_screen/android/src/main/kotlin/com/builttoroam/dual_screen/DualScreenPlugin.kt

```diff
@@ -0,0 +1,132 @@
+        activity = this.activity
+        this.context = context
+        MethodChannel(messenger, METHOD_CHANNEL_NAME).setMethodCallHandler(this)
+        EventChannel(messenger, EVENT_CHANNEL_NAME).setStreamHandler(this)
+    }
+
+    override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        this.registerPlugin(binding.getApplicationContext(), binding.getBinaryMessenger())
+    }
+
+    companion object {
+        @JvmStatic
+        fun registerWith(registrar: Registrar) {
+            DualScreenPlugin().registerPlugin(registrar.context(), registrar.messenger())
+        }
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "isDualScreenDevice" -> {
+                result.success(isDualScreenDevice())
+            }
+            "isAppSpanned" -> {
+                if (isDualScreenDevice()) {
+                    result.success(isAppSpanned())
+                } else {
+                    result.success(false)
+                }
+            }
+            else -> {
+                result.notImplemented()
+            }
+        }
+    }
+
+    override fun onListen(arguments: Any?, events: EventSink?) {
+        activity?.getWindow()?.getDecorView()?.getRootView()?.addOnLayoutChangeListener(object : OnLayoutChangeListener {
+            override fun onLayoutChange(
+                v: View?,
+                left: Int,
+                top: Int,
+                right: Int,
+                bottom: Int,
+                oldLeft: Int,
+                oldTop: Int,
+                oldRight: Int,
+                oldBottom: Int
+            ) {
+                events!!.success(isAppSpanned())
+            }
+        })
+    }
+
+    override fun onCancel(arguments: Any?) {
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        this.activity = null
+    }
+
+    override fun onDetachedFromActivity() {
+        this.activity = null
+    }
+
+    fun isDualScreenDevice(): Boolean {
+        return context?.getPackageManager()!!.hasSystemFeature(
+            "com.microsoft.device.display.displaymask"
+        )
+    }
+
+    fun isAppSpanned(): Boolean {
+        if (isDualScreenDevice()) {
+            var boundings = DisplayMask.fromResourcesRectApproximation(this.activity).getBoundingRects()
+            if (boundings.isEmpty()) {
+                return false
+            }
+            var drawingRect = android.graphics.Rect()
+            activity?.getWindow()?.getDecorView()?.getRootView()?.getDrawingRect(drawingRect)
+            return boundings.get(0).intersect(drawingRect)
```

Comment:
Why do we use `0` here? We should create a variable that will describe its purpose or make a comment. I would suggest to go with a variable / const approach.

Reply:
I've added a comment, i think the const approach is a bit unnecessary

Reply:
I disagree. Random variables used in code (like this one) don't have any meaning and are a pain to decypher when reading and understanding code. We should always try to make the code as much readable as possible.

It might be that to make this code more readable we should use different approach and assign the `boundings.get(0)` to a variable with a name that would describe the boundries element, instead of extracing 0 to a const.

Reply:
@nickrandolph which approach would you like to take?

Reply:
Is there not a "first()" method that we can use instead of get(index)

Reply:
There is! pushed a changed to use first()